be/src/exprs/jit/jit_expr.cpp

```diff
@@ -0,0 +1,104 @@
+// Copyright 2021-present StarRocks, Inc. All rights reserved.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     https://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+#include "exprs/jit/jit_expr.h"
+
+#include <chrono>
+#include <vector>
+
+#include "column/chunk.h"
+#include "common/compiler_util.h"
+#include "common/status.h"
+#include "exprs/anyval_util.h"
+#include "exprs/expr.h"
+#include "exprs/function_context.h"
+#include "exprs/jit/jit_functions.h"
+#include "llvm/IR/IRBuilder.h"
+
+namespace starrocks {
+
+JITExpr* JITExpr::create(ObjectPool* pool, Expr* expr) {
+    TExprNode node;
+    node.node_type = TExprNodeType::JIT_EXPR;
+    node.opcode = TExprOpcode::JIT;
+    node.is_nullable = expr->is_nullable();
+    node.type = expr->type().to_thrift();
+    node.output_scale = expr->output_scale();
+    node.is_monotonic = expr->is_monotonic();
+    return pool->add(new JITExpr(pool, node, expr));
+}
+
+JITExpr::JITExpr(ObjectPool* pool, const TExprNode& node, Expr* expr) : Expr(node), _pool(pool), _expr(expr) {
+    _expr->get_uncompilable_exprs(_children);
+}
+
+Status JITExpr::prepare(RuntimeState* state, ExprContext* context) {
+    Status status = Status::OK();
+    RETURN_IF_ERROR(Expr::prepare(state, context));
+
+    _mem_pool = std::make_unique<MemPool>();
+
+    FunctionContext::TypeDesc return_type = AnyValUtil::column_type_to_type_desc(type());
+    std::vector<FunctionContext::TypeDesc> args_types;
+    for (Expr* child : _children) {
+        args_types.push_back(AnyValUtil::column_type_to_type_desc(child->type()));
+    }
+    _fn_context_index = context->register_func(state, return_type, args_types);
+
+    FunctionContext* fn_ctx = context->fn_context(_fn_context_index);
+
+    // TODO(Yueyang): remove this time cost.
+    auto start = std::chrono::high_resolution_clock::now();
+    // Compile the expression into native code and retrieve the function pointer.
+    auto function = JITFunction::compile_scalar_function(_expr);
+    auto end = std::chrono::high_resolution_clock::now();
+    std::chrono::duration<double> elapsed = end - start;
+    if (function != nullptr) {
+        LOG(INFO) << "Yueyang: JIT compile success, time cost: " << elapsed.count() << " s";
+    } else {
+        LOG(INFO) << "Yueyang: JIT compile failed, time cost: " << elapsed.count() << " s";
+    }
+
+    auto function_state = _pool->add(new JITFunctionState{function});
+
+    fn_ctx->set_function_state(FunctionContext::FunctionStateScope::THREAD_LOCAL, function_state);
```

Comment:
may be should `FRAGMENT_LOCAL` instead of `THREAD_LOCAL`, as the jit expr are shared used for all drivers (fragment instance) in each thread.